Expose internals used with other style guides

IndrajeetPatil opened the pull request:
Re-implementation of abandoned #1037, which had complex merge conflicts.

Closes #974.

# Progress tracker

- [x] Export `compute_parse_data_nested()` helper
- [x] Export `is_*()` helpers
- [x] Add examples to new exports
- [x] Check function and argument names are okay (since these are now user-facing)
- [x] Code clean-up
- [x] ~~Update NEWS~~ (`{fledge}` is supposed to take care of this)
- [x] Code review

IndrajeetPatil committed:
Expose internals used with other style guides

github-actions[bot] committed:
pre-commit

IndrajeetPatil committed:
minor

Robinlovelace commented:
Hi @IndrajeetPatil just following-up on this, is there any way I can help progress? Will be great to be able to have `styler.*` packages building on this!

IndrajeetPatil committed:
Add examples for remaining new exports

IndrajeetPatil committed:
Example for `compute_parse_data_nested()`

IndrajeetPatil committed:
Update compute_parse_data_nested.Rd

IndrajeetPatil commented on R/expr-is.R:
```diff
@@ -1,38 +1,68 @@
+#' pd <- compute_parse_data_nested(code)
+#' is_curly_expr(pd)
+#' is_curly_expr(pd$child$`17`$child$`14`)
+#'
```
The blank lines are deliberate. 

Without them, all examples are squished together like a centipede, and it's hard to tell where one ends and the other starts.

IndrajeetPatil commented on R/expr-is.R:
```diff
@@ -44,56 +74,68 @@ is_function_call <- function(pd) {
+#' is_function_declaration(pd$child$`12`$child$`11`)
+#'
+#' @export
+is_function_declaration <- function(pd) {
```
I think this is a better name than `is_function_dec()`.

IndrajeetPatil commented on R/expr-is.R:
```diff
@@ -1,38 +1,68 @@
+#' is_conditional_expr(pd$child$`24`)
+#'
+#' @export
+is_conditional_expr <- function(pd) {
```
I think this is a better name than `is_cond_expr()`.

lorenzwalthert replied:
I think we should keep it as is (or change all the `*_expr()` functions).

IndrajeetPatil replied:
But why do we need to change all the `*_expr()` functions? Their current names are expressive as they are.

lorenzwalthert replied:
no sorry I read your comment wrongly. I thouhg you wanted `*_expression()` everywhere, but that is (obviously) not the case.

IndrajeetPatil replied:
I see.

Okay, then this should be ready for your review. I don't have any additional changes in mind.

IndrajeetPatil commented:
Hi @Robinlovelace, I am done on my end. Now I await code review. Once that's done, I think this should be good to go and might even be part of the next release (cf. #1044).

Robinlovelace commented:
That is a great update, thanks. :crossed_fingers: for the next release!

lorenzwalthert commented:
@IndrajeetPatil I saw the review request. Please do not re-request a review from me. I don't forget about them, but I don't have time to look into it now but I will so in due time.

lorenzwalthert committed:
reference via index, not name

this should be more robust against changes in the R parser with the goal to avoid CRAN submission due to failing examples

lorenzwalthert commented:
@IndrajeetPatil  can you have a look at my last commit? After this one is approved by you, I will prepare the release as discussed in #1044.

IndrajeetPatil commented:
> can't request a review from you for some reason

The one who made PR can't review it themselves. That's why you can't assign me to review my own PR.

> can you have a look at my last commit 

Your changes look good to me. Thanks.
